src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -1239,6 +1265,16 @@ def limit(self, n: int, *, offset: int = 0) -> "SelectStatement":
 
         return new
 
+    def with_subqueries(self, subquery_plans: List[SnowflakePlan]) -> "SelectStatement":
+        super().with_subqueries(subquery_plans)
+        for plan in subquery_plans:
```

Comment:
i see that other than the selectStatement, and setStatement. All rest just get the schema query from the attached snowflake plan. SetStatement also just get the attribute from the child snowflake plan directly. Therefore, probably only SelectStatement need a special handling here. Please comment it clearly, the code are quite confusing here for people do not understand how code generation works completely

Reply:
I actually think this is not needed for selectables at all. I wrote a summary here https://github.com/snowflakedb/snowpark-python/pull/2434#discussion_r1823386898 but if you notice, `schema_query` for `SelectStatement` is always made using `from_.schema_query`. In case of `SnowflakePlan`, default `schema_query` is made from given query so the adjustment is required.